third_party/move/move-compiler/src/parser/syntax.rs

```diff
@@ -2253,10 +2251,30 @@ fn parse_dot_or_index_chain(context: &mut Context) -> Result<Exp, Box<Diagnostic
             Tok::Period => {
                 context.tokens.advance()?;
                 let n = parse_identifier_or_positional_field(context)?;
-                let ahead = context.tokens.peek();
-                if matches!(ahead, Tok::LParen | Tok::ColonColon) {
-                    let generics = if ahead == Tok::ColonColon {
-                        context.tokens.advance()?;
+                let has_generics = match context.tokens.peek() {
+                    Tok::Less if n.loc.end() as usize == context.tokens.start_loc() => true,
+                    Tok::ColonColon
+                        if context.tokens.lookahead().unwrap_or(Tok::ColonColon) == Tok::Less =>
+                    {
+                        let loc = context.tokens.advance_with_loc()?;
+                        if context.env.flags().language_version() >= LanguageVersion::V2_2 {
+                            context.env.add_diag(diag!(
+                                Uncategorized::DeprecatedWillBeRemoved,
+                                (
+                                    loc,
+                                    format!(
+                                        "The `::` qualifier is obsolete since Move {}",
```

Comment:
We need a test that emits this error.

Reply:
I had one, but then other tests in this folder don't work any longer, which require language v2.1. I'd prefer we leave it like this and we will get the error once we set the default to 2.2